[Bug]: T5 prepare_decoder_input_ids_from_labels return None

### 软件环境

```Markdown
- paddlepaddle:
- paddlepaddle-gpu: 2.3.1.post101
- paddlenlp: latest
```


### 重复问题

- [X] I have searched the existing issues

### 错误描述

```Markdown
[here](https://github.com/PaddlePaddle/PaddleNLP/blob/255360ffcfb9637a01667311e642a6e8d867694e/examples/information_extraction/DuUIE/uie/seq2struct/data_collator.py#L506)，self.model.prepare_decoder_input_ids_from_labels方法返回None。

type(self.model)
<class 'paddlenlp.transformers.t5.modeling.T5ForConditionalGeneration'>
getattr(self.model, "prepare_decoder_input_ids_from_labels")
getattr(self.model, "prepare_decoder_input_ids_from_labels") is None
True
```


### 稳定复现步骤 & 代码

![image](https://user-images.githubusercontent.com/11239531/210475997-67f09d9e-603b-4602-862c-ed02c0e43a10.png)

会在`_shift_right`这里遇到一个断言错误，即`bos_token_id is not None`。
报错信息：
```bash
AssertionError: bos_token_id has to be defined. In T5 it is usually set to the pad_token_id. See T5 docs for more information

```
[相关代码](https://github.com/PaddlePaddle/PaddleNLP/blob/bd2be0f97484776e4a8c03630b6dfe6932fdd5d6/paddlenlp/transformers/t5/modeling.py#L686)

--- a/paddlenlp/transformers/t5/modeling.py
+++ b/paddlenlp/transformers/t5/modeling.py
@@ -1553,6 +1553,9 @@ class T5ForConditionalGeneration(T5PretrainedModel):
             "use_cache": use_cache,
         }
 
+    def prepare_decoder_input_ids_from_labels(self, labels: paddle.Tensor):
+        return self._shift_right(labels)
+
     @staticmethod
     def expand_inputs_for_generation(input_ids, expand_size, attention_mask=None, **model_kwargs):
         index = paddle.tile(paddle.arange(input_ids.shape[0]).unsqueeze(-1), [1, expand_size]).reshape([-1])
@@ -1650,6 +1653,7 @@ class T5EncoderModel(T5PretrainedModel):
         output_hidden_states: Optional[bool] = None,
         return_dict: Optional[bool] = None,
     ):
+
         return_dict = return_dict if return_dict is not None else self.config.use_return_dict
         encoder_outputs = self.encoder(
             input_ids=input_ids,

